Release v7.4.0 bug fixes

cmark opened the pull request.

Commit:
Test plugins optional.

apeteri committed:
[snomed] Allow external configuration of namespace-module assigners

cmark committed:
[config] remove unnecessary systemUser configuration key

cmark committed:
[api] fix bug details() expansion affectedComponentId filtering

Fixes #473.

cmark committed:
[core] fix RevisionSegment.withEnd(newEnd) logic

Do not extend the segment to later timestamps than the currently set end
value. Fixes branch@timestamp scenarios where multiple parent branches
are participating in the search.

cmark committed:
[index] increase embedded Elasticsearch max_buckets to 1.500.000...

...temporarily

A proper fix for large aggregations using composite agg is planned for
v7.5.0.

apeteri commented on core/com.b2international.snowowl.identity/src/com/b2international/snowowl/identity/domain/User.java:
```diff
@@ -86,6 +86,7 @@ public boolean isAdministrator() {
 	 * @return
 	 */
 	public boolean hasPermission(Permission permissionRequirement) {
+		if (true) return true;
```
Please remove diagnostic modifications before merging.

cmark committed:
Revert "[core] fix RevisionSegment.withEnd(newEnd) logic"

This reverts commit 7db86acac27f5c65241dd01650b6ebb46a0410de.

cmark committed:
[core] use new RevisionSegment.restrictEnd(restrictTo) method...

...to restrict segments to certain timestamp value.

Fixes branch@timestamp scenarios where multiple parent branches with
shorter segments are participating in the search.

cmark committed:
[core] remove diagnostic code statement
